Engine/Component/ComponentTransform.h

```diff
@@ -39,6 +41,8 @@ class ComponentTransform : public Component
 
 	float4x4 model_matrix = float4x4::identity;
 	float4x4 global_model_matrix = float4x4::identity;
+
+	friend class ComponentCamera;
```

Comment:
Why using friend?

Reply:
We want to bind in `ComponentCamera` frustum position (lets call it **FP**) and the owner transform position (lets call it **TP**). Considering that there are a lot of methods in `ComponentCamera` that modify FP (`Orbit`, `MoveXXX`, ...), I approached the problem in this way:
1. In `ComponentCamera` the only position updated is TP.
2. In every frame, I set FP with TP values.
3. I generate corresponding camera matrices.

With this approach I ensure that TP == FP and also that changes to TP outside `ComponentCamera` are also reflected in FP. But this approach has the problem of accessing the private member `translation` of `ComponentTransform`.

So, to solve that I befriended `ComponentCamera` with `ComponentTransform` class. Another possible solution could be using Getters/Setters in `ComponentCamera`.

Reply:
You created GetTranslation, can't you use it?

Reply:
I guess :D I'm on it!

![](https://media.giphy.com/media/JIX9t2j0ZTN9S/giphy.gif)